wire-ios/Wire-iOS/Sources/Helpers/syncengine/Conversation+Participants.swift

```diff
@@ -37,67 +34,63 @@ extension ZMConversation {
         return maxParticipants - 1
     }
 
-    func addOrShowError(participants: [UserType]) {
-        guard let session = ZMUserSession.shared(),
-                session.networkState != .offline else {
-            self.showAlertForAdding(for: NetworkError.offline)
-            return
+    func removeOrShowError(participant user: UserType, completion: ((VoidResult) -> Void)? = nil) {
+
+        @Sendable func fail(with error: Error) {
+            showAlertForRemoval(for: error)
+            completion?(.failure(error))
```

Comment:
It's probably out of scope of this PR, but I think we should factor away methods like this from managed objects. I see there's a use case for adding participants, why not one for removing participants?

Reply:
There's one for removing participants but I haven't pushed it yet. I was waiting for feedback from @netbe before completing the implementation and starting to write tests

Reply:
I also agree it's a bit out of the scope and I was considering making this in another PR

Reply:
Will be done here https://wearezeta.atlassian.net/browse/WPB-5137